torch/__init__.py

```diff
@@ -88,7 +88,7 @@ def _running_with_deploy():
     from .version import cuda as cuda_version
     import glob
     if cuda_version and all(not glob.glob(os.path.join(p, 'cudart64*.dll')) for p in dll_paths):
-        cuda_version_1 = cuda_version.replace('.', '_')
+        cuda_version_1 = str(cuda_version).replace('.', '_')
```

Comment:
Maybe better to make cuda_version a string before line 90?

Reply:
actually, this is idiomatic, if cuda_version is None --> if cuda_version would not be True if cuda_version is None

so I did object conversion under cuda_version evaluation to True condition


actually, this is draft - but look for some domain knowledge if 12.1 should be supported experimentally and under-the-table only, or from nightly indeed